File: CommonRecycler/src/main/java/com/jcodecraeer/xrecyclerview/base/BaseRefreshInterface.java
package com.jcodecraeer.xrecyclerview.base;

/**
 * Created by guoshuyu on 2017/1/7.
 */

interface BaseRefreshInterface {

    void onMove(float delta);

    void refreshComplete();

    void setProgressStyle(int style);

    void setArrowImageView(int resid);

    boolean releaseAction();

    int getVisibleHeight();

    int getState();

    void setVisibleHeight(int height);

    void setState(int state);

    void reset();
}

File: CommonRecycler-Kotlin/src/main/java/com/shuyu/commonrecycler/xrecycler/base/BaseRefreshInterface.kt
package com.shuyu.commonrecycler.xrecycler.base

/**
 * Created by guoshuyu on 2017/1/7.
 */

open  interface BaseRefreshInterface {

    var visibleHeight: Int

    var state: Int

    fun onMove(delta: Float)

    fun refreshComplete()

    fun setProgressStyle(style: Int)

    fun setArrowImageView(resid: Int)

    fun releaseAction(): Boolean

    fun reset()
}
